packages/vscode-extension/src/debugging/ReconnectingDebugSession.ts

```diff
@@ -0,0 +1,118 @@
+import { Disposable } from "vscode";
+import { Cdp } from "vscode-cdp-proxy";
+import { Metro } from "../project/metro";
+import { CancelToken } from "../utilities/cancelToken";
+import { sleep } from "../utilities/retry";
+import { DebugSession, DebugSource, JSDebugConfiguration } from "./DebugSession";
+import { Devtools } from "../project/devtools";
+import { disposeAll } from "../utilities/disposables";
+
+const PING_TIMEOUT = 1000;
+export class ReconnectingDebugSession implements DebugSession, Disposable {
+  private disposables: Disposable[] = [];
+  private cancelReconnect: CancelToken | undefined;
+
+  private isRunning: boolean = false;
+
+  constructor(
+    private readonly debugSession: DebugSession & Partial<Disposable>,
+    private readonly metro: Metro,
+    devtools: Devtools
+  ) {
+    this.disposables.push(debugSession.onDebugSessionTerminated(this.maybeReconnect));
+    this.disposables.push(devtools.onEvent("appReady", this.maybeReconnect));
+  }
+
+  public async startJSDebugSession(configuration: JSDebugConfiguration) {
+    this.isRunning = true;
+    return this.debugSession.startJSDebugSession(configuration);
+  }
+
+  private async pingJsDebugSessionWithTimeout() {
+    const resultPromise = this.debugSession
+      .evaluateExpression({ expression: "('ping')" })
+      .then((response) => {
+        return response.result.value === "ping";
```

Comment:
```suggestion
        return true;
```

I think it would be just a bit cleaner (can you ever get a repsonse and not return true?